provisioning/transport/mqtt/src/ProvisioningChannelHandlerAdapter.cs

```diff
@@ -326,6 +333,29 @@ private async Task VerifyPublishPacketTopicAsync(IChannelHandlerContext context,
             }
         }
 
+        private static TimeSpan? getRetryAfterFromTopic(string topic)
+        {
+            string[] topicAndQueryString = topic.Split('?');
```

Comment:
This is complex parsing - please add a UT.

Reply:
Sure thing

Reply:
This method is buried deep enough into internal classes and methods that unit testing it without mocks doesn't seem feasible. Would you be okay with using reflection to isolate this method for unit testing?

Reply:
I factored out this logic to the ProvisioningErrorDetails implementations for each protocol in order to make this logic unit testable